python/cuml/tests/test_sklearn_compatibility.py

```diff
@@ -80,18 +134,641 @@
+        "check_estimator_sparse_matrix": "HDBSCAN does not handle sparse matrices gracefully",
+        "check_parameters_default_constructible": "HDBSCAN parameters are mutated on init",
+        "check_estimators_pickle": "HDBSCAN does not support pickling",
+        "check_estimators_pickle(readonly_memmap=True)": "HDBSCAN does not support pickling with readonly memmap",
+        "check_f_contiguous_array_estimator": "HDBSCAN does not handle F-contiguous arrays",
+        "check_methods_sample_order_invariance": "HDBSCAN results depend on sample order",
+        "check_methods_subset_invariance": "HDBSCAN results depend on data subset",
+        "check_fit2d_1sample": "HDBSCAN does not handle single sample properly",
+        "check_fit2d_1feature": "HDBSCAN does not handle single feature properly",
+        "check_dict_unchanged": "HDBSCAN modifies input dictionaries",
+        "check_fit_idempotent": "HDBSCAN fit is not idempotent",
+        "check_fit_check_is_fitted": "HDBSCAN does not check is_fitted properly",
+        "check_n_features_in": "HDBSCAN does not set n_features_in properly",
+        "check_fit1d": "HDBSCAN does not raise ValueError for 1D input",
+        "check_fit2d_predict1d": "HDBSCAN does not handle 1D prediction input gracefully",
+    },
 }
 
 
 def get_xfails(estimator):
     return PER_ESTIMATOR_XFAIL_CHECKS.get(type(estimator), {})
 
 
+def _check_name(check):
+    if hasattr(check, "__wrapped__"):
+        return _check_name(check.__wrapped__)
+    return (
+        check.func.__name__ if isinstance(check, partial) else check.__name__
+    )
+
+
 @estimator_checks.parametrize_with_checks(
-    [KMeans(), LogisticRegression()], expected_failed_checks=get_xfails
+    [
+        KernelRidge(),
+        GaussianNB(),
+        ComplementNB(),
+        CategoricalNB(),
+        BernoulliNB(),
+        MultinomialNB(),
+        UMAP(),
+        TSNE(),
+        TruncatedSVD(),
+        IncrementalPCA(),
+        PCA(),
+        SVR(),
+        SVC(),
+        LinearSVR(),
+        LinearSVC(),
+        NearestNeighbors(),
+        KNeighborsRegressor(),
+        KNeighborsClassifier(),
+        KernelDensity(),
+        Ridge(),
+        ElasticNet(),
+        Lasso(),
+        LinearRegression(),
+        RandomForestClassifier(),
+        RandomForestRegressor(),
+        KMeans(),
+        LogisticRegression(),
+    ],
+    expected_failed_checks=get_xfails,
 )
 def test_sklearn_compatible_estimator(estimator, check):
     # Check that all estimators pass the "common estimator" checks
     # provided by scikit-learn
 
+    # These estimators lead to additional MemoryErrors in the other
+    # estimators. As a result they are currently skipped.
+    if isinstance(
+        estimator,
+        (GaussianNB, ComplementNB, CategoricalNB, BernoulliNB, MultinomialNB),
+    ):
+        pytest.skip(
+            "Estimator leads to additional MemoryErrors in other estimators"
```

Comment:
Please make sure to reference relevant issues where possible. This goes for here and other instances.